### PR TITLE
Support property naming strategy for Guava Ranges

### PR DESCRIPTION
**Pull request by philleonard**
Follow up PR for #56. Unsure if the public constructors are better off with a different overload structure to preserve backwards compatibility.

### CODE REVIEW

File: guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/RangeSerializer.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 import com.google.common.collect.BoundType;
 import com.google.common.collect.Range;
+import java.util.Optional;
```
Comment (cowtowncoder) [inline guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/RangeSerializer.java]: note: JDK 8, can not use for Jackson 2.x

**Comment by cowtowncoder** [comment]
I merged unit test part (well, second of tests) in 2.10, to help verify, merge the fix.

**Comment by philleonard** [comment]
Thanks @cowtowncoder. I have fixed it to transform the names in `createContextual` and have removed the JDK 8 usages. LMK if there is a better way to do this.

**Comment by cowtowncoder** [comment]
@philleonard Looks good, I think, any tweaks I can think I can do post-merge I think. The only other thing is that I'd like to get this in 2.10, will see what'd be the best way.

Just one other thing before I can merge this: if I have not yet asked for CLA (https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf), I would need that now, before the first contribution. After I receive filled one I can merge any and all contributions across projects, so it is just one time hassle.
The most common way to do this is to print, fill  & sign, scan (cell phone photo is fine), email to `info` at fasterxml dot com.

Looking forward to merging this fix, thank you again for contributing it!

**Comment by philleonard** [comment]
Thanks, contributor agreement has been sent your way.

**Comment by cowtowncoder** [comment]
Thank you again -- I merged this, and will merge in 2.10, do some light touch up.

In future it might be neat to simplify the process too, but first things first.

**Pull request merged**